addons/mrp/models/mrp_production.py

```diff
@@ -296,8 +296,12 @@ def _compute_components_availability(self):
         other_productions.components_availability_state = False
 
         all_raw_moves = productions.move_raw_ids
-        # Force to prefetch more than 1000 by 1000
-        all_raw_moves._fields['forecast_availability'].compute_value(all_raw_moves)
+        
+        # Force to prefetch more than 1000 by 1000, only for records not in cache.
+        forecast_missing_ids = set(self.env.cache.get_missing_ids(all_raw_moves, all_raw_moves._fields['forecast_availability']))
+        all_raw_moves_without_forecast = all_raw_moves.filtered(lambda m: m.id in forecast_missing_ids)
```

Comment:
Can't you turn the `all_raw_moves_without_forecast.ids` into a set then browse the diff ? I'm not so sure that it's way more efficient than a filtered, but since `productions.move_raw_ids` can be quite large I'm afraid you might have some regressions with `filtered`.

Reply:
Good point, I believe it would be faster? I can do a quick benchmark for this just bc I'm curious how it'll do

Reply:
@Aurelienvd 
OK so I'm probably going to end up closing this because I cannot reproduce the compute function being called twice anymore.. In the customer's database or runbot.

But regarding this comment I did do a neat benchmark in a fresh 17 DB that proves your point true:

Running this is ~4 seconds
```py
from datetime import datetime

ps = env['res.partner'].browse(range(10_000_000))
sub_ps = set(range(1_000_000))

ts = datetime.now()
result = ps.filtered(lambda p: p.id in sub_ps)
te = datetime.now()
print(te - ts)
```

Whereas this is only ~0.7 seconds:
```py
from datetime import datetime

ps = env['res.partner'].browse(range(10_000_000))
sub_ps = set(range(1_000_000))

ts = datetime.now()
ps_set = set(ps.ids)
result = env['res.partner'].browse(ps_set & sub_ps)
te = datetime.now()
print(te - ts)
```


(Note I did need to do set union as opposed to diff, bc we want the IDs from both of them - but yeah I think that's what you meant originally)

Hope this is insightful :)